### PR TITLE
Add F12 Quit window.

### PR DESCRIPTION
**Pull request by fmatthew5876**
When you hold F12 a window appears and counts down. Once
countdown is complete, we reset to title screen.

Replaces 2k3e Alt+F12 feature. Enabled for all versions of the engine.

Also fixes a crash with F12, we capture old_scene before
checking for reset.

### CODE REVIEW

File: src/game_quit.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with EasyRPG Player. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef EP_GAME_QUIT_H
+#define EP_GAME_QUIT_H
+
+#include "window_help.h"
+
+class Game_Quit {
```
Comment (Ghabry) [inline src/game_quit.h]: Indent for public/private must be one level to the left

File: src/game_quit.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	public:
+		Game_Quit();
+		void Update();
+
+		bool ShouldQuit() const;
+
+		void Enable(bool value);
+	private:
+		void Reset();
+	private:
```
Comment (Ghabry) [inline src/game_quit.h]: private twice

**Comment by fmatthew5876** [comment]
> Also makes now Steam more happy, as F12 is the screenshot key.

Except now anytime you try to create a stream screenshot you'll have this window

> An observable difference is that reset is now not working on Title and Load (from title) anymore but that's fine for me.

I was back and forth on this. But since it's supported in 2k3 I added it back. Also the title load menu kind of makes sense.

I also verified it does not appear in game browser.

> How about reset 2 seconds? 3 seconds feels a bit long and when TestPlay is enabled maybe instant as I use reset alot when testing and this breaks my Workflow https://xkcd.com/1172/ ;)

Changed to 2 seconds

**Comment by Ghabry** [comment]
> Except now anytime you try to create a stream screenshot you'll have this window

There is a short delay after F12 until Steam creates a screenshot, so this should work.

File: src/window_base.h [hunk 3]
```diff
@@ -40,7 +40,7 @@ class Window_Base : public Window {
 	 * @param width window width.
 	 * @param height window height.
 	 */
```
Comment (carstene1ns) [inline src/window_base.h]: dox.

Comment (fmatthew5876) [reply src/window_base.h]: Fixed

**Comment by fmatthew5876** [comment]
I see what you mean about it being handy to not have this feature be too slow. In later PRs I changed it so fast forward affects the speed of the F12 window.

**Comment by Ghabry** [comment]
I just noticed what you wrote. Which PR made it affected by FF?

**Comment by fmatthew5876** [comment]
#2017 the main loop refactor makes keeping this out of ff tricky. Also, after using it a lot now I think it's better that it does get sped up. It's still not too fast to make a mistake and makes testing easier.

**Comment by Ghabry** [comment]
yeah I WANT it in FF because I'm already annoyed xD

**Comment by carstene1ns** [comment]
Good thing we mostly run in test play mode 🤣

**Pull request merged**